src/mlpack/methods/ann/gan/metrics/frechet_distance.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file frechet_distance.hpp
+ * @author Toshal Agrawal
+ *
+ * Definition of Frechet Inception Distance for Generative Adversarial Networks.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_GAN_METRICS_FRECHET_DISTANCE_HPP
+#define MLPACK_METHODS_ANN_GAN_METRICS_FRECHET_DISTANCE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+
+/**
+* Function that computes Frechet Inception Distance for a set of images
+* produced by a GAN.
+*
+* For more information, see the following.
+*
+* For more information, see the following paper:
+*
+* @code
+* @article{
+*   author    = {Martin Heusel, Hubert Ramsauer, Thomas Unterthiner, Bernhard
+*                Nessler, Sepp Hochreiter},
+*   title     = {GANs Trained by a Two Time-Scale Update Rule Converge to a
+*                Local Nash Equilibrium},
+*   year      = {2017},
+*   url       = {https://arxiv.org/abs/1706.08500},
+*   eprint    = {1706.08500},
+* }
+* @endcode
+*/
+class FrechetDistance
```

Comment:
Can you add a test for the metric?

Reply:
Yes, sure. Actually, this is still in work in progress. In the author's implementation `inception model's layer3` values are used. So I was thinking to try that.  I will add a test once I get to know whether it is possible or not to get those values. I am also looking at the closed inception layer PR to know whether it is possible to implement.

Reply:
Sounds good, let me know if you need any help.